packages/typespec-client-generator-core/test/types/multipart-types.test.ts

```diff
@@ -61,6 +61,68 @@ describe("typespec-client-generator-core: multipart types", () => {
       code: "@azure-tools/typespec-client-generator-core/conflicting-multipart-model-usage",
     });
   });
+  it("multipart conflicting model usage for only multipart operations", async function () {
+    await runner.compile(
+      `
+        @service({title: "Test Service"}) namespace TestService;
+        model Address {city: string;}
+        model MultiPartRequest {
+          address: Address;
+          id: string;
+          profileImage: bytes;
+        }
+        
+        @post
+        @route("/basic1") 
+        op basic1(@header contentType: "multipart/form-data", @body body: MultiPartRequest): NoContentResponse;
+        @post
+        @route("/basic2") 
+        op basic2(@header contentType: "multipart/form-data", @body body: MultiPartRequest): NoContentResponse;
+      `
+    );
+    deepEqual(runner.context.diagnostics.length, 0);
+    const address = runner.context.sdkPackage.models.find((x) => x.name === "Address");
+    ok(address);
+    deepEqual(address.usage, UsageFlags.Input);
```

Comment:
Since `usage.Json` is calculated recursively from regular body. If it is not used in regular body, the model usage will not be assigned to `Json`.